src/manifest/__tests__/filter_supported_representations.test.ts

```diff
@@ -0,0 +1,138 @@
+  beforeEach(() => {
+    jest.resetModules();
+  });
+
+  it("should not filter text adaptations", () => {
+    const isCodecSupportedSpy = jest.fn(() => false);
+    jest.mock("../../compat", () => ({ isCodecSupported: isCodecSupportedSpy }));
+
+    const filterSupportedRepresentations = require("../filter_supported_representations")
+      .default;
+
+    const rep1 = { id: "54", bitrate: 100 };
+    const rep2 = { id: "55", bitrate: 101 };
+    expect(filterSupportedRepresentations("text", [rep1, rep2]))
+      .toEqual([rep1, rep2]);
+
+    expect(isCodecSupportedSpy).not.toHaveBeenCalled();
+  });
+
+  it("should not filter image adaptations", () => {
+    const isCodecSupportedSpy = jest.fn(() => false);
+    jest.mock("../../compat", () => ({ isCodecSupported: isCodecSupportedSpy }));
+
+    const filterSupportedRepresentations = require("../filter_supported_representations")
+      .default;
+
+    const rep1 = { id: "54", bitrate: 100 };
+    const rep2 = { id: "55", bitrate: 101 };
+    expect(filterSupportedRepresentations("image", [rep1, rep2]))
+      .toEqual([rep1, rep2]);
+
+    expect(isCodecSupportedSpy).not.toHaveBeenCalled();
+  });
+
+  it("should filter video adaptations based on the MIME type and codec", () => {
+    const isCodecSupportedSpy = jest.fn((arg : string) => arg.indexOf("webm") !== -1);
+    jest.mock("../../compat", () => ({ isCodecSupported: isCodecSupportedSpy }));
+
+    const filterSupportedRepresentations = require("../filter_supported_representations")
+      .default;
+
+    const rep1 = { id: "54", bitrate: 100, mimeType: "video/mp4", codecs: "avc2" };
+    const rep2 = { id: "55", bitrate: 101, mimeType: "video/webm", codecs: "vp9" };
+    expect(filterSupportedRepresentations("video", [rep1, rep2]))
+      .toEqual([rep2]);
+
+    expect(isCodecSupportedSpy).toHaveBeenCalledTimes(2);
+    expect(isCodecSupportedSpy).toHaveBeenCalledWith("video/mp4;codecs=\"avc2\"");
+    expect(isCodecSupportedSpy).toHaveBeenCalledWith("video/webm;codecs=\"vp9\"");
+  });
+
+  it("should filter audio adaptations based on the MIME type and codec", () => {
+    const isCodecSupportedSpy = jest.fn((arg : string) => arg.indexOf("aac") !== -1);
+    jest.mock("../../compat", () => ({ isCodecSupported: isCodecSupportedSpy }));
+
+    const filterSupportedRepresentations = require("../filter_supported_representations")
+      .default;
+
+    const rep1 = { id: "54", bitrate: 100, mimeType: "audio/mp4", codecs: "aac" };
+    const rep2 = { id: "55", bitrate: 101, mimeType: "audio/webm", codecs: "ogg" };
+    expect(filterSupportedRepresentations("video", [rep1, rep2]))
+      .toEqual([rep1]);
+
+    expect(isCodecSupportedSpy).toHaveBeenCalledTimes(2);
+    expect(isCodecSupportedSpy).toHaveBeenCalledWith("audio/mp4;codecs=\"aac\"");
+    expect(isCodecSupportedSpy).toHaveBeenCalledWith("audio/webm;codecs=\"ogg\"");
+  });
+
+  it("should filter audio adaptations based on the MIME type and codec", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 69. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=440&issues=AWgpL7_hUMkE2J58feAe&open=AWgpL7_hUMkE2J58feAe)

[ISSUE_KEY]: AWgpL7_hUMkE2J58feAe